src/main/java/org/molgenis/vcf/inheritance/matcher/GenmodCompoundMapper.java

```diff
@@ -0,0 +1,74 @@
+package org.molgenis.vcf.inheritance.matcher;
+
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFFileReader;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GenmodCompoundMapper {
+
+  private static final String COMPOUNDS_INFO_FIELD = "Compounds";
+
+  Map<String, Set<String>> createVariantGeneList(VCFFileReader fileReader,
+      VepMapper vepMapper) {
+    Map<String, Set<String>> result = new HashMap<>();
+    for (VariantContext vc : fileReader) {
+      String key = String
+          .format("%s_%d_%s_%s", vc.getContig(), vc.getStart(), vc.getReference().getBaseString(),
+              vc.getAlternateAlleles().stream().map(Allele::getBaseString)
+                  .collect(Collectors.joining(",")));
+      Set<String> genes = vepMapper.getGenes(vc);
+      result.put(key, genes);
+    }
+    return result;
+  }
+
+  //roundabout way of getting the actual string from the infofield, which is malformed in case of multi allelics
```

Comment:
can we prevent this malformed input? is this introduced by genmod? in that case I suggest we notify them of this issue.

Reply:
Cannot be prevented, deep down in the docs there is mention of a "split alleles" flag, but it is not available in the release. I will create an issue on their git repo...